tutorials/mmvnn.py

```diff
@@ -0,0 +1,76 @@
+r"""
+MMV with non-negative norms
+===========================
+In this tutorial we will use the ``spg_mmv`` solver which solves a
+multi-measurement vector basis pursuit denoise problem. Both standard L12
+and L12non-negative norms will be compared. The latter set of norms is used in
+case allowsour solution to have only positive values in case we have access to
+such a kind of prior information.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+from spgl1 import spg_mmv
+from spgl1 import _norm_l12nn_primal, \
+    _norm_l12nn_dual, _norm_l12nn_project
```

Comment:
Wait - why are these underscore functions if we're expecting other people to use them? That's not generally good practice. Did I do this to begin with?

Reply:
Totally agree. I did this with the default ones as they were not meant to be used (or at least not explicitly) and did the same here but since we expect people to use those as replacement I will take away the _ and add a commit :)

Reply:
I just made a new commit where all non-negative norms are now public methods. 

L1 non-negative norms also used to be private. This meant that since they were not used directly in the library itself, none could have used them (course someone could have violated the _ 'convention' and use them... but that would have been bad of them). So now those methods are all public and the private ones are those that are used as defaults norms as i don't see why people should use them separately. Hope you agree with this change :)